core/src/com/ericc/the/game/map/Dungeon.java

```diff
@@ -0,0 +1,152 @@
+package com.ericc.the.game.map;
+
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.EntityListener;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.badlogic.gdx.math.GridPoint2;
+import com.ericc.the.game.Engines;
+import com.ericc.the.game.Mappers;
+import com.ericc.the.game.Media;
+import com.ericc.the.game.components.DescendingComponent;
+import com.ericc.the.game.components.MobComponent;
+import com.ericc.the.game.components.PlayerComponent;
+import com.ericc.the.game.components.PositionComponent;
+import com.ericc.the.game.entities.Mob;
+import com.ericc.the.game.entities.Player;
+import com.ericc.the.game.entities.PushableObject;
+import com.ericc.the.game.entities.Stairs;
+import com.ericc.the.game.helpers.Moves;
+import com.sun.istack.internal.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * This class stores the entire dungeon, level by level.
+ * By level here we mean the map.
+ */
+public class Dungeon {
+    private ArrayList<Map> levels;
+    private ArrayList<ArrayList<Entity>> entities;
+    private int currentLevel;
+    private Engines engines;
+
+    public Dungeon(Engines engines) {
+        this.engines = engines;
+        this.entities = new ArrayList<>();
+
+        levels = new ArrayList<>();
+        currentLevel = -1;
+    }
+
+    /**
+     * Changes the current Map to the next one if it exists,
+     * if not it does create one.
+     */
+    public Map goToNext() {
+        if (levels.isEmpty() || currentLevel + 1 == levels.size()) {
+            addNewMap();
+        }
+
+        if (currentLevel >= 0) {
+            saveLastProgress();
+        }
+
+        ++currentLevel;
+
+        if (currentLevel < entities.size()) {
```

Comment:
`currentLevel < entities.size()`

co tu się...? :oo

Reply:
dobra, już zrozumiałem. Bo to nie jest entities, tylko entitiesPerLevel

Reply:
Sorki Filip, ale nazewnictwo serio ma znaczenie, bo przez 10 minut wpatrywałem się w ten kod i próbowałem zrozumieć czemu ilość ilość entities na obecnym poziomie ma się równać jego numerkowi xD